gemma/ops.h

```diff
@@ -92,6 +92,43 @@ HWY_INLINE constexpr size_t RowsPerStrip() {
   return kRowsPerStrip;
 }
 
+HWY_INLINE void ToEvenOddF32(
+    const hwy::bfloat16_t* HWY_RESTRICT vec_aligned, const size_t size,
+    float* HWY_RESTRICT out) {
+  const hn::ScalableTag<float> df;
+  const hn::Repartition<hwy::bfloat16_t, decltype(df)> dbf16;
+  const hn::RebindToUnsigned<decltype(df)> du32;
+  const auto odd = Set(du32, 0xFFFF0000u);
+  using VF32 = decltype(hn::Zero(df));
+
+  HWY_DASSERT(size % hn::Lanes(dbf16) == 0);
+  HWY_DASSERT(hn::IsAligned(df, vec_aligned));
+
+  VF32 veven, vodd;
+  for (size_t i = 0; i < size; i += hn::Lanes(dbf16)) {
+    Bf16ToF32EO(df, vec_aligned + i, veven, vodd);
+    hn::Store(veven, df, out + i);
+    hn::Store(vodd, df, out + i + hn::Lanes(df));
+  }
+}
+
+HWY_INLINE void ToEvenOddF32(
```

Comment:
Is it possible to avoid calling this for f32 inputs? Seems like we'd be better off doing a direct f32*f32 instead.

Reply:
Got it, thanks. I now understand why this is useful even for already-f32 vectors.